chainspec/src/main.rs

```diff
@@ -0,0 +1,48 @@
+extern crate serde_json;
+extern crate serde_ignored;
+extern crate ethjson;
+
+use std::collections::BTreeSet;
+use std::{fs, env, process};
+use ethjson::spec::Spec;
+
+fn quit<S: AsRef<str>>(s: S) -> ! {
```

Comment:
I think that `Foo: AsRef<Bar>` as a type parameter is unnecessary because of Rust's automatic coercion (for example, `foo(&bar)` will succeed if `foo` takes a `&T` and `bar` implements `AsRef<T>`). I think I've made this same mistake in my latest PR with `AsRef<Path>`. So this could just take `&str` with the same API.